src/components/images/draw.js

```diff
@@ -132,8 +133,21 @@ module.exports = function draw(gd) {
         var ya = Axes.getFromId(gd, d.yref);
 
         var size = fullLayout._size;
-        var width = xa ? Math.abs(xa.l2p(d.sizex) - xa.l2p(0)) : d.sizex * size.w;
-        var height = ya ? Math.abs(ya.l2p(d.sizey) - ya.l2p(0)) : d.sizey * size.h;
+        var width, height;
+        if(xa !== undefined) {
+            width = ((typeof(d.xref) === 'string') && d.xref.endsWith(' domain')) ?
```

Comment:
I thought I saw a comment about this earlier but perhaps not... `endsWith` is not supported by IE. Can we use `Axes.getRefType` for these instead?

Reply:
In fact you use `Axes.getRefType` down below (line 201 & 2) so you may as well just call it once up above where you get `xa` and `ya`. In fact given how it's used you could probably just make booleans `xIsDomain` and `yIsDomain`